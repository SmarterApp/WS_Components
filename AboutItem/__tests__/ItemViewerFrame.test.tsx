import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemFrame } from '../ItemViewerFrame';
import * as TestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';

describe("ItemViewerFrame", () => {
    const obj = {
        url: "http://test.com",
        loading:  false
    }

    it("matches snapshot", () => {
        expect(shallow(<ItemFrame {...obj}/>)).toMatchSnapshot();;
    })
})