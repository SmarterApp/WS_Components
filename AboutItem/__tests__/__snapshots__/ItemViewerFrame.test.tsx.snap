// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemViewerFrame matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="itemViewerFrame"
    tabIndex={0}
>
    <div
        className="itemviewer-iframe-spinner"
    >
        <img
            src="/images/spin-large.gif"
        />
    </div>
    <iframe
        className="itemviewer-iframe"
        id="itemviewer-iframe"
        onLoad={[Function]}
        onLoadStart={[Function]}
        src="http://test.com"
    />
</div>,
  "nodes": Array [
    <div
      className="itemViewerFrame"
      tabIndex={0}
>
      <div
            className="itemviewer-iframe-spinner"
      >
            <img
                  src="/images/spin-large.gif"
            />
      </div>
      <iframe
            className="itemviewer-iframe"
            id="itemviewer-iframe"
            onLoad={[Function]}
            onLoadStart={[Function]}
            src="http://test.com"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ItemFrame
        loading={false}
        url="http://test.com"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ItemFrame {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "finishLoad": [Function],
        "props": Object {
          "loading": false,
          "url": "http://test.com",
        },
        "refs": Object {},
        "startLoad": [Function],
        "state": Object {
          "loading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="itemViewerFrame"
          tabIndex={0}
>
          <div
                    className="itemviewer-iframe-spinner"
          >
                    <img
                              src="/images/spin-large.gif"
                    />
          </div>
          <iframe
                    className="itemviewer-iframe"
                    id="itemviewer-iframe"
                    onLoad={[Function]}
                    onLoadStart={[Function]}
                    src="http://test.com"
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="itemViewerFrame"
          tabIndex={0}
>
          <div
                    className="itemviewer-iframe-spinner"
          >
                    <img
                              src="/images/spin-large.gif"
                    />
          </div>
          <iframe
                    className="itemviewer-iframe"
                    id="itemviewer-iframe"
                    onLoad={[Function]}
                    onLoadStart={[Function]}
                    src="http://test.com"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ItemFrame
    loading={false}
    url="http://test.com"
/>,
}
`;
